### PR TITLE
feat: deploy the coding assistant application in vscode

### PR DESCRIPTION
**Pull request by bm424**
* Removes the recipe view (Completes SOU-1746)
* Removes the review view
* Adds the built javascript and styles from the coding assistant application
* Tidy up of types and names, including namespacing the commands and requests

**Note:** as this workflow stabilises, I'd like to ensure the changes to the webview script and styles are committed automatically into the frontend. For now, I think it's fine to keep it manual.

Review alongside sourcery-ai/core#2982

### CODE REVIEW

File: src/chat.ts [hunk 1]
```diff
@@ -111,23 +118,38 @@ export class ChatProvider implements vscode.WebviewViewProvider {
...
+          }
+          case "cancelRequest": {
             vscode.commands.executeCommand("sourcery.chat_cancel_request");
             break;
           }
-          case "open_link_request": {
+          case "recipeRequest": {
+            vscode.commands.executeCommand("sourcery.chat_request", request);
+            break;
+          }
```
Comment (brendanator) [inline src/chat.ts:143]: **🎯 suggestion:** ‏Remove all request and command specific code from src/chat.ts and src/extension.ts and move all logic into the webapp

*Goal* - make it so that when we add new messages from or to the webapp that we don't need to change the vscode (and therefore the pycharm) extensions. This has a few benefits:
- We only need to change the front and backend code, not the extensions. Right now the message also needs adding to chat.ts and extension.ts, as well as their PyCharm equivalents
- Additionaliy this means we only need to change a single repo to make a change, not three
- This will allow us to develop and test more quickly

One possible solution is that there's just one inbound message and one outbound message and these are routed directly through to the webapp / binary via the plugins. If the webapp needs extra info from the IDE it can use functions on the injected IDE interface.

Open to other solutions if you have them

*Note* - this can be done in follow up PRs

**Review by brendanator — APPROVED** [review]
Awesome!

I have tested this out and everything seems to be working perfectly 🥂 

I have one important comment about how to adjust generalise the messaging code, but that can come in follow up work

A few comments on this screenshot:
- The Clear button can be removed at the very top right
- There's now an expander panel immediately inside the Coding Assistant which looks funny
- The font size in the inner expander is larger than in the outer expander
- The message paper aeroplane icon has no colour
- The recipes do not currently have a blue background
- The file in the chat that is a link has no colour
image

File: src/chat.ts (hunk 1 above)
Comment (bm424) [reply src/chat.ts:143]: Yeah I think this is going to be a good optimisation (I did more or less exactly your single-message suggestion for the troubleshooting view and it worked very well) and I'll add it as a change to make once the development has stabilised a bit. 

That said, I think there are actually two types of outbound message: ones for the server and ones for the extension (I've tried to codify this in the new type names). Neither is entirely transparent through the extension. The server ones will usually have additional data attached, such as the text selection, open files, and so on - this is information we don't have direct access to in the web app. We could pass it in to the web app (setting up relevant event listeners) or attach it on outbound messages - either seems reasonable. The extension ones will be used for e.g. the link handlers, direct text edits, and so on. We could skip these if we're happy for it to be a round trip to the server, but it's unclear exactly what the set of these IDE interactions is likely to be in the short or medium term, and routing through the server is getting a bit convoluted.

Basically all I'm saying is I don't want to jump too quickly into making this improvement before we have a better idea of the scope of it, but I agree with the principle.

Comment (bm424) [reply src/chat.ts:143]: This is how I see where we should be aiming. To your point, there should be no direct app -> extension -> app loops.

image

Comment (brendanator) [reply src/chat.ts:143]: This all makes sense, and the distinction between server and extension requests makes a lot of sense

> The server ones will usually have additional data attached, such as the text selection, open files, and so on - this is information we don't have direct access to in the web app. We could pass it in to the web app (setting up relevant event listeners) or attach it on outbound messages

I think there's a third alternative which is probably cleaner. Rather than use event listeners to pass that information to the webapp, the webapp can request it from the extension through the injected IDE API

> Basically all I'm saying is I don't want to jump too quickly into making this improvement before we have a better idea of the scope of it, but I agree with the principle

Yep, fair enough. I would really like this before putting the webapp into PyCharm, so let's scope this out fully in our 1-1

File: src/chat.ts [hunk 2]
```diff
@@ -174,186 +227,133 @@ export class ChatProvider implements vscode.WebviewViewProvider {
...
   private async _getHtmlForWebview(webview: vscode.Webview) {
     // Get the local path to main script run in the webview, then convert it to a uri we can use in the webview.
-    const scriptUri = webview.asWebviewUri(
-      vscode.Uri.joinPath(this._extensionUri, "src", "webview", "chat.js")
-    );
-
-    // Do the same for the stylesheet.
-    const styleResetUri = webview.asWebviewUri(
-      vscode.Uri.joinPath(this._extensionUri, "media", "reset.css")
+    const baseSrc = webview.asWebviewUri(
```
Comment (brendanator) [inline src/chat.ts:295]: **🎯 suggestion:** ‏Rename some of these variables

It's not clear what is what with some of these. Are they ide style/interface implementations or the main coding assistant webapp:
- baseSrc
- scriptSrc
- bridgeSrc
- styleSheetSrc

**Review by brendanator — APPROVED** [review]
👍

File: src/chat.ts (hunk 2 above)
Comment (bm424) [reply src/chat.ts:298]: I've renamed some of these and added comments to help clarify.
